pkg/kn/flags/podspec_test.go

```diff
@@ -114,31 +115,31 @@ func TestPodSpecResolve(t *testing.T) {
 		Use: "test",
 		Run: func(cmd *cobra.Command, args []string) {
 			podSpec := &corev1.PodSpec{Containers: []corev1.Container{{}}}
-			err := flags.ResolvePodSpec(podSpec, cmd.Flags())
+			err := flags.ResolvePodSpec(podSpec, cmd.Flags(), inputArgs)
 			assert.NilError(t, err, "PodSpec cannot be resolved.")
 			assert.DeepEqual(t, expectedPodSpec, *podSpec)
 		},
 	}
-	testCmd.SetArgs(args)
+	testCmd.SetArgs(inputArgs)
 	flags.AddFlags(testCmd.Flags())
 	testCmd.Execute()
 }
 
 func TestPodSpecResolveReturnError(t *testing.T) {
 	outBuf := bytes.Buffer{}
 	flags := &PodSpecFlags{}
+	inputArgs := []string{"--mount", "123456"}
 	testCmd := &cobra.Command{
 		Use: "test",
 		Run: func(cmd *cobra.Command, args []string) {
 			podSpec := &corev1.PodSpec{Containers: []corev1.Container{{}}}
-			err := flags.ResolvePodSpec(podSpec, cmd.Flags())
+			err := flags.ResolvePodSpec(podSpec, cmd.Flags(), inputArgs)
 			fmt.Fprint(cmd.OutOrStdout(), "Return error: ", err)
 		},
 	}
 	testCmd.SetOut(&outBuf)
 
-	args := []string{"--mount", "123456"}
-	testCmd.SetArgs(args)
+	testCmd.SetArgs(inputArgs)
```

Comment:
@zroubalik did you try to mock `os.Args` in here to help with various unit tests that currently fails?

https://github.com/knative/client/pull/1328/checks?check_run_id=2656432279#step:6:27

Reply:
Yeah, I just wanted to make this testcase consistent with the others. These tests are okay, the problem is in those, where we don't invoke `ResolvePodSpec()` directly, but not using the command line for the actual invocaton. As I mentioned in the PR description.